sdk/search/azure-search-documents/swagger/src/main/java/SearchServiceCustomizations.java

```diff
@@ -476,4 +638,136 @@ private static void bulkRemoveMethod(PackageCustomization packageCustomization,
+
+        // Update the Javadocs.
+        javadoc.getBlockTags().set(returnIndex, new JavadocBlockTag(JavadocBlockTag.Type.RETURN,
+            String.format("the %s object itself.", returnType)));
+        javadoc.addBlockTag(new JavadocBlockTag(JavadocBlockTag.Type.THROWS, String.format(
+            "IllegalArgumentException If {@code %s} is supplied when {@link #getSkillVersion()} is {@link %s#%s}.",
+            setterVariable, versionType, version)));
+        method.setJavadocComment(javadoc);
+
+
+        BlockStmt body = method.getBody().get();
+        body.getStatements().add(0, createAndIfThrowStatement("IllegalArgumentException",
+            String.format("%s using %s doesn't support '%s'.", returnType, version, setterVariable),
+            new BinaryExpr(new NameExpr(setterVariable), new NullLiteralExpr(), BinaryExpr.Operator.NOT_EQUALS),
+            new BinaryExpr(new NameExpr("version"), versionExpression, BinaryExpr.Operator.EQUALS)));
+
+        clazz.addMember(method);
+    }
+
+    /**
+     * Creates an if-throw statement.
+     *
+     * <pre>
+     * if (a != null && b == c) {
+     *     throw logger.logExceptionAsError(new Exception(exceptionMessage));
+     * }
+     * </pre>
+     *
+     * Is what this method creates.
+     *
+     * @param exceptionType Type of the exception.
+     * @param exceptionMessage Message of the exception.
+     * @param check1 First statement in the conditional.
+     * @param check2 Second statement in the conditional.
+     * @return A new throw statement.
+     */
+    private static IfStmt createAndIfThrowStatement(String exceptionType, String exceptionMessage,
+        BinaryExpr check1, BinaryExpr check2) {
+        // Create check1 && check2
+        BinaryExpr conditional = new BinaryExpr(check1, check2, BinaryExpr.Operator.AND);
+
+        // Create throw block in the if.
+        BlockStmt throwBlock = new BlockStmt(new NodeList<>(new ThrowStmt(
+            createLogAndThrowCall("logger", "logExceptionAsError",
+                createExceptionWithMessage(exceptionType, exceptionMessage)))));
+
+        return new IfStmt(conditional, throwBlock, null);
+    }
+
+    private static MethodCallExpr createLogAndThrowCall(String loggerName, String logMethod,
+        Expression exceptionExpression) {
+        return new MethodCallExpr(new NameExpr(loggerName), logMethod, new NodeList<>(exceptionExpression));
+    }
+
+    private static ObjectCreationExpr createExceptionWithMessage(String exceptionType, String exceptionMessage) {
+        return new ObjectCreationExpr().setType(exceptionType).setArguments(new NodeList<>(
+            new StringLiteralExpr(exceptionMessage)));
+    }
+
+    private static Javadoc createJavadoc(String description, JavadocBlockTag... tags) {
+        Javadoc javadoc = new Javadoc(JavadocDescription.parseText(description));
+        for (JavadocBlockTag tag : tags) {
+            javadoc.addBlockTag(tag);
+        }
+
+        return javadoc;
+    }
```

Comment:
Do we have any plans for Search to minimize these customizations? There's a lot of customization happening here and we should look into which ones can be moved to swagger.

Reply:
Yes, we should look at some AutoRest configurations to lessen the number of customization. I think the two quick wins would be configuration of abstract and final classes that overrides and inference that AutoRest currently uses